spec/controllers/password_resets_controller_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+
+describe PasswordResetsController do
+  describe "GET show" do
+    it "renders show template if token is valid" do
+      alice = Fabricate(:user)
+      alice.update_column(:token, '123')
+      alice.save
+      get :show, id:'123'
+      expect(response).to render_template :show
+    end
+
+    it "sets @token" do
+      alice = Fabricate(:user)
+      alice.update_column(:token, '123')
+      alice.save
+      get :show, id:'123'
+      expect(assigns(:token)).to eq('123')
+    end
+
+    it "redirects to expired token page if token is invalid" do
+      get :show, id: 'wrongtoken123'
+      expect(response).to redirect_to expired_token_path
+    end
+  end
+
+  describe "POST create" do
+    context "with valid token" do
+      it "redirects to the sign in page" do
+        alice = Fabricate(:user, password: 'old_password')
+        alice.update_column(:token, '123')
```

Comment:
I had taken the other route: of creating tokens only when resetting password and be nil otherwise. Instead of updating the column, I now have
```
alice = Fabricate(:user, password: 'old_password', token: '123')
```

thank you :+1: